src/pages/patientView/PatientViewPage.tsx

```diff
@@ -350,14 +361,17 @@ export default class PatientViewPage extends React.Component<IPatientViewPagePro
                                 <MutationInformationContainer
                                     mutations={this.state.mutationData}
                                     hotspots={this.state.hotspotsData}
-                                    mrnaExprRankData={patientViewPageStore.mrnaExprRankData.isComplete ? patientViewPageStore.mrnaExprRankData.result : undefined }
+                                    mrnaExprRankData={ patientViewPageStore.mrnaExprRankData }
                                     mutSigData={this.state.mutSigData}
                                     sampleOrder={sampleManager.sampleOrder}
                                     sampleLabels={sampleManager.sampleLabels}
                                     sampleColors={sampleManager.sampleColors}
                                     sampleTumorType={mockData.tumorType}
                                     sampleCancerType={mockData.cancerType}
                                     sampleManager={ sampleManager }
+                                    onVisibleRowsChange={ patientViewPageStore.requestMrnaExprRankData.isComplete
+                                    ? (data:Mutation[][]) => this.lazyFetchVisibleRowData(data, patientViewPageStore.requestMrnaExprRankData.result)
```

Comment:
I think we can check/access the store inside of the lazyFetch handler so as to keep this cleaner.   To achieve functional purity, we can have another layer of indirection there.